carbonmark-api/src/utils/helpers/listings.utils.ts

```diff
@@ -35,6 +36,11 @@ const isActiveListing = (
   Number(l.leftToSell) >= (minSupply || 0) &&
   Number(l.leftToSell) > 0;
 
+export const getListingById = async (sdk: GQL_SDK, id: string) => {
+  const listing = await sdk.marketplace.getListingById({ id });
+  return listing;
+};
+
```

Comment:
If you are making that util function, maybe you can put all the logic that enables it to transform the GQL response into a proper formatted listing. We may want to reuse that.

```    
    if (!listing) {
      return null;
    }

    const symbol = (await getTokenById(sdk, listing.tokenAddress)).symbol;

    const formattedListing = formatListing(listing);

    return {
      ...formattedListing,
      symbol,
    };

```

Reply:
done in [06cea64](https://github.com/KlimaDAO/klimadao/pull/2046/commits/06cea64fca48faa90d66c3932bf539442413f4a2)